### ISSUE TITLE
Filter by Attribute > Update the logic for the Filter by Attribute with the or (Any) condition.

### ISSUE THREAD

**Issue by nefeline** [woothemes/woocommerce#44746]
On https://github.com/woocommerce/woocommerce-blocks/discussions/9023#discussioncomment-8450521 we kicked off a discussion regarding what the logic for filtering attributes with the **Or (Any)** condition should look like, and the outcome of that discussion has been the following (please feel free to correct and/or update this issue in case I missed anything, @jarekmorawski!):

1. Whenever using the or (Any) condition, the attribute counts should not be changed at all when the user selects multiple values within the same attribute (e.g. when selecting multiple colors). 
2. Whenever different attributes are selected (e.g. a color attribute in combination with a size attribute), the counts are updated to match the returned results.
3. Whenever different attributes are selected (e.g. blue (color) and large (size)), any products that don't match the criteria, returning a count equal to zero, should not be displayed on the filters list.

The reasoning for this is as follows (porting Jarek's comment from https://github.com/woocommerce/woocommerce-blocks/discussions/9023#discussioncomment-8450521 ):

> This is the piece we've been missing. If attribute values are always connected with an OR operator, they're never exclusive and the counts shouldn't be relative. It's like buying groceries: putting 5 apples in your basket doesn't mean you can only buy 1 orange.

> Our general goal for the counters should be to illustrate how many products will be added to the product list after applying a filter. If someone's viewing an unfiltered list, activating the first filter will tell them the total number of results. Activating another one will let them add X more. Deactivating it will remove X.

**Comment by gigitux** [comment]
@roykho would you mind verifying if this issue is still relevant with the new filter blocks?

**Comment by roykho** [comment]
@gigitux I don't believe this is relevant to the new filter blocks. I just tested it and it already does everything listed here.

**Comment by gigitux** [comment]
@roykho given that we are looking to deprecate the legacy filter blocks (https://github.com/woocommerce/woocommerce/issues/55614), I think that we can close this issue: WDYT?
